File: backend/payment/src/main/java/com/bbmovie/payment/service/zalopay/ZaloPayConstraint.java
package com.bbmovie.payment.service.zalopay;

@SuppressWarnings("all")
public class ZaloPayConstraint {
    public static final String ONLY_SUPPORTED_CURRENCY = "VND";

    // Endpoints
    public static final String CREATE_ORDER_URL_SANDBOX_V1 = "https://sandbox.zalopay.com.vn/v001/tpe/createorder";
    public static final String CREATE_ORDER_URL_PROD_V1 = "https://zalopay.com.vn/v001/tpe/createorder";

    public static final String CREATE_ORDER_URL_SANDBOX_V2 = "https://sb-openapi.zalopay.vn/v2/create";
    public static final String CREATE_ORDER_URL_PROD_V2 = "https://openapi.zalopay.vn/v2/create";

    // Callback field names for ZaloPay webhook
    public static final String CALLBACK_DATA = "data";
    public static final String CALLBACK_MAC = "mac";

    // Vietnam timezone for apptransid prefix (yymmdd)
    public static final String VIETNAM_TZ = "Asia/Ho_Chi_Minh";
}

File: backend/payment/src/main/kotlin/com/bbmovie/payment/service/payment/zalopay/ZaloPayConstraint.kt
package com.bbmovie.payment.service.payment.zalopay

object ZaloPayConstraint {
    const val ONLY_SUPPORTED_CURRENCY = "VND"

    // Endpoints
    const val CREATE_ORDER_URL_SANDBOX = "https://sandbox.zalopay.com.vn/v001/tpe/createorder"
    const val CREATE_ORDER_URL_PROD = "https://zalopay.com.vn/v001/tpe/createorder"

    // Callback field names for ZaloPay webhook
    const val CALLBACK_DATA = "data"
    const val CALLBACK_MAC = "mac"

    // Vietnam timezone for apptransid prefix (yymmdd)
    const val VIETNAM_TZ = "Asia/Ho_Chi_Minh"
}


